### PR TITLE
Add/update scripts to allow for pre-commit and pre-push git hooks

### PR DESCRIPTION
**Pull request by randy-concepcion**
- Add scripts to lint on pre-commits and test on pre-pushes
- Add collectCoverageFrom for jest to include untested files
- Add global test coverage thresholds
- Add husky package
- Exclude folders for the todo script
- Add documentation for pre-commit and pre-push git hooks
- Upgrade Jest to latest to used 'changedSince' option
- Update snapshots due to upgrade: see facebook/jest#1681

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -29,6 +31,17 @@ steps:
     artifact_paths:
       - "coverage/lcov.info"
 
+  # Run tests without coverage if this is a hotfix
+  # This is necessary for builds triggered by PR mergest to master
```
Comment (danielnovograd) [inline .buildkite/pipeline.yml]: **merges

**Comment by randy-concepcion** [comment]
@danielnovograd Thanks for the review! Made the following changes:
- Fixed `merges` spelling error
- Updated `/coverage/` folder in `.gitignore`

**Comment by randy-concepcion** [comment]
Did an overhaul of the `pre-commit` and `pre-push` git hook scripts.
- Commits were only linting JS/JSX files. Needed to add SCSS and MD linting too.
- Pushes were auto-fixing lint errors after the commit was pushed to remote. Now, the pre-push commit will do the following:
    - Prevent pushes from happening when lint errors are auto-fixed, and
    - List the files which were auto-fixed so it can be reviewed, committed properly, and included in the same push

Even though the pre-commit hooks will trigger linters by default, this can be overridden with the `--no-verify` option, e.g., `git commit -m "why u no lint?" --no-verify`.

**NOTE**: The `--no-verify` flag can be aliased, e.g, `git cnv`, where `cnv = commit --no-verify` in the `~/.gitconfig` file.

**Comment by randy-concepcion** [comment]
Closing stale PR.

**Pull request closed without merging**